batch/batch/batch.py

```diff
@@ -147,27 +150,116 @@ def batch_record_to_dict(app, record):
+    log_store = app['log_store']
+
+    batch_id = record['batch_id']
+    job_id = record['job_id']
+    format_version = record['format_version']
+
+    if format_version == 1:
+        return record['spec']
+
+    bunch_record = await db.select_and_fetchone(
+        '''
+SELECT token, start_job_id FROM batch_bunches
+WHERE batch_id = %s AND start_job_id <= %s
+ORDER BY start_job_id DESC
+LIMIT 1;
+''',
+        (batch_id, job_id))
+    token = bunch_record['token']
+    start_job_id = bunch_record['start_job_id']
+
+    try:
+        return await log_store.read_spec_file(batch_id, token, start_job_id, job_id)
+    except google.api_core.exceptions.NotFound:
+        return None
+
 
-    attributes = spec.pop('attributes', None)
+async def _get_full_job_status(app, record):
+    log_store = app['log_store']
+
+    batch_id = record['batch_id']
+    job_id = record['job_id']
+    attempt_id = record['attempt_id']
+    state = record['state']
+    format_version = record['format_version']
+
+    if state in ('Pending', 'Ready', 'Cancelled'):
+        return
+
+    if state in ('Error', 'Failed', 'Success'):
+        if format_version == 1:
+            return json.loads(record['status'])
+
+        try:
+            status = await log_store.read_status_file(batch_id, job_id, attempt_id)
+            return json.loads(status)
+        except google.api_core.exceptions.NotFound:
+            log.exception(f'missing status file for {id}')
+            return None
+
+    assert state == 'Running'
+    assert record['status'] is None
+
+    ip_address = record['ip_address']
+    async with aiohttp.ClientSession(
+            raise_for_status=True, timeout=aiohttp.ClientTimeout(total=60)) as session:
+        try:
+            url = (f'http://{ip_address}:5000'
+                   f'/api/v1alpha/batches/{batch_id}/jobs/{job_id}/status')
+            resp = await request_retry_transient_errors(session, 'GET', url)
+            return await resp.json()
+        except aiohttp.ClientResponseError as e:
+            if e.status == 404:
+                return None
+            raise
+
+
+async def job_record_to_dict(app, record, include_spec_status=True):
+    format_version = record['format_version']
+
+    status = record['status']
+    if status:
+        status = json.loads(status)
+        if format_version == 1:
+            status = {'status': status}
+            exit_code = Job.exit_code(status)
+            duration = humanize_timedelta_msecs(Job.total_duration_msecs(status))
+        else:
+            exit_code = status.get('exit_code')
+            duration = humanize_timedelta_msecs(status.get('duration'))
+    else:
+        exit_code = None
+        duration = None
 
     result = {
         'batch_id': record['batch_id'],
         'job_id': record['job_id'],
         'state': record['state'],
-        'spec': spec
+        'exit_code': exit_code,
+        'duration': duration
     }
 
+    spec = json.loads(record['spec'])
```

Comment:
And same for the spec?  Are these schemas of these commented anywhere?  For example, there is a comment in worker.py on the function that produces that job status.